lib/active_model/serializer.rb

```diff
@@ -191,10 +191,24 @@ def json_key
     end
 
     def read_attribute_for_serialization(attr)
+      # Start by favoring serializers method override
       if respond_to?(attr)
         send(attr)
       else
-        object.read_attribute_for_serialization(attr)
+        # Otherwise, check read_attribute_for_serialization
+        if object.respond_to?(:read_attribute_for_serialization)
+          object.read_attribute_for_serialization(attr)
```

Comment:
This is a *huge* philosophical break and also greatly expands the surface area that AMS needs to cover from a defined interface to a defined interface when available, else try to guess.  I don't think we should support serializing primitives such as Hashes.